spec/formula_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper.rb'
+
+describe Statsample::GLM::Formula do
+  context '#parse_formula' do
+    context 'no interaction' do
+      let(:formula) { described_class.new 'y ~ a+b' }
+      subject { formula.parse_formula }
+      
+      it { is_expected.to be_a Array }
+      its(:first) { is_expected.to eq Statsample::GLM::Token }
+      its(:to_basic_formula) { is_expected.to eq 'y ~ 1+a(-)+b(-)' }
```

Comment:
@zverok Ok. Let me make it fully working first, then you can see all the details and then we can come up with a better design.